File: December - 27/KishoreMuruganantham_signal_propagation_network.R
input <- file("stdin", "r")
lines <- readLines(input)
n <- as.integer(lines[1])
m <- as.integer(lines[2])

adj <- vector("list", n)
if (m > 0) {
  for (i in 1:m) {
    vals <- as.numeric(unlist(strsplit(lines[2 + i], "\\s+")))
    u <- vals[1] + 1
    v <- vals[2] + 1
    w <- vals[3]
    adj[[u]] <- c(adj[[u]], list(c(v, w)))
  }
}

s_Line <- lines[2 + m + 1]
s_node <- as.integer(s_Line) + 1

dist <- rep(Inf, n)
dist[s_node] <- 0
visited <- rep(FALSE, n)

for (count in 1:n) {
  min_val <- Inf
  u <- -1
  
  for (i in 1:n) {
    if (!visited[i] && dist[i] < min_val) {
      min_val <- dist[i]
      u <- i
    }
  }
  
  if (u == -1) break
  visited[u] <- TRUE
  
  neighbors <- adj[[u]]
  if (length(neighbors) > 0) {
    for (nb in neighbors) {
      v_idx <- nb[1]
      wt <- nb[2]
      if (dist[u] + wt < dist[v_idx]) {
        dist[v_idx] <- dist[u] + wt
      }
    }
  }
}

res <- max(dist)
if (res == Inf) {
  cat("-1\n")
} else {
  cat(res, "\n", sep = "")
}

File: December - 27/KishoreMuruganantham_signal_propagation_network.swift
import Foundation
var lines = [String]()
while let line = readLine() { lines.append(line) }
if !lines.isEmpty {
    let n = Int(lines[0]) ?? 0
    let m = Int(lines[1]) ?? 0
    
    struct Edge { let to: Int, w: Double }
    var adj = [[Edge]](repeating: [], count: n)
    
    if m > 0 {
        for i in 1...m {
            let parts = lines[1 + i].split(separator: " ").compactMap { Double($0) }
            let u = Int(parts[0])
            let v = Int(parts[1])
            let w = parts[2]
            adj[u].append(Edge(to: v, w: w))
        }
    }
    
    let sNodeIdx = (lines.count > 2 + m) ? Int(lines[2 + m]) ?? 0 : 0
    
    var dist = [Double](repeating: Double.infinity, count: n)
    dist[sNodeIdx] = 0
    var visited = [Bool](repeating: false, count: n)
    
    for _ in 0..<n {
        var minVal = Double.infinity
        var u = -1
        
        for i in 0..<n {
            if !visited[i] && dist[i] < minVal {
                minVal = dist[i]
                u = i
            }
        }
        
        if u == -1 { break }
        visited[u] = true
        
        for e in adj[u] {
            if dist[u] + e.w < dist[e.to] {
                dist[e.to] = dist[u] + e.w
            }
        }
    }
    
    if let res = dist.max() {
        if res == Double.infinity {
            print("-1")
        } else {
            if floor(res) == res {
                 print(Int(res))
            } else {
                 print(res)
            }
        }
    } else {
        print("-1")
    }
}
